videoretrieval/base/base_dataset.py

```diff
@@ -46,4 +47,75 @@ def download_dataset(self):
     @abstractmethod
     def download_and_cache_precomputed_features(self):
         """Downloads and caches precomputed features for the given dataset."""
-        pass
\ No newline at end of file
+        pass
+
+    @property
+    @abstractmethod
+    def video_captions():
+        """Returns a dict of that maps from video_id to a list of captions."""
+        pass
+
+    @property
+    @abstractmethod
+    def train_valid_test_ids(self):
+        """Returns a tuple of sets providing ids for the dataset splits.
+
+        Returns: a tuple of sets, where the first set contains the ids for the 
+        train data, the second for the validation data, and the third for the
+        test data."""
+
+        pass
+
+    def build_generator(self, data):
+        for example in data:
+            yield example
+
+    def build_id_caption_pair_generator_dataset(self, data):
+        generator = lambda: self.build_generator(data)
+
+        return tf.data.Dataset.from_generator(generator, (tf.string, tf.string))
+    
```

Comment:
These are internal and simple functions which is why there aren't docstrings (per the google open source style guide).